pkg/agent/openflow/pipeline.go

```diff
@@ -586,26 +592,77 @@ func (f *featurePodConnectivity) podClassifierFlow(podOFPort uint32, isAntreaFle
 
 // podUplinkClassifierFlows generates the flows to mark the packets with target destination MAC address from uplink/bridge
 // port, which are needed when uplink is connected to OVS bridge and Antrea IPAM is configured.
-func (f *featurePodConnectivity) podUplinkClassifierFlows(dstMAC net.HardwareAddr) []binding.Flow {
+func (f *featurePodConnectivity) podUplinkClassifierFlows(dstMAC net.HardwareAddr, vlanID uint16) []binding.Flow {
 	cookieID := f.cookieAllocator.Request(f.category).Raw()
-	return []binding.Flow{
+	var flows []binding.Flow
+	ctZoneTypeRegMark := IPCtZoneTypeRegMark
+	if len(f.ipProtocols) == 1 && f.ipProtocols[0] == binding.ProtocolIPv6 {
+		ctZoneTypeRegMark = IPv6CtZoneTypeRegMark
```

Comment:
Do you mean ctZoneTypeRegMark is same for IPv4 only and dual-stack, but different for IPv6 only? Why not use a single value if it is already used for dual-stack case?

Reply:
No, this function support 3 cases.
IPv4 only: Match related packets and set `IPCtZoneTypeRegMark`.
IPv6 only: Match related packets and set `IPv6CtZoneTypeRegMark`.
DS: Match related IPv6 packets and set `IPv6CtZoneTypeRegMark`, and then match related packets and set `IPCtZoneTypeRegMark`.

Reply:
> Another question is how do you expect to process APR packets from uplink/br-int? I don't think conntrack supports ARP, so if ARP with VLAN also enters into OVS pipeline from uplink/br-int, I would prefer a seperate flow.

As changed by Flexible Pipeline, ARP stages have higher priority, thus ARP packet won't be passed to this table.

Reply:
@wenyingd The two questions have same answer:
We cannot guarantee that only IP and IPv6 packet will be passed here. Thus we need 3 flows for each in_port:
Common part:

```
				MatchInPort(port).
				MatchDstMAC(dstMAC).
				MatchVLAN(nonVLAN, vlanID, nil).
				Action().LoadToRegField(VLANIDField, uint32(vlanID)).
				Action().LoadRegMark(FromUplinkRegMark).
				Action().GotoStage(stageConntrackState)
```

1. Match IP, then LoadRegMark(IPCtZoneTypeRegMark)
2. Match IPv6, then LoadRegMark(IPv6CtZoneTypeRegMark)
3. Default

For current implementation, we combined 1+3 for IP only and DS, 2+3 for IPv6 only, to reduce the flow count. And it is harmless to set IPCtZoneTypeRegMark/IPv6CtZoneTypeRegMark for non-IP/IPv6 packet.
If we only accept IP/IPv6 for these flows, your suggestion will work.

Reply:
`ClassifierTable` is for IP/IPv6 packet, it is the first table for IP/IPv6 pipeline.

Reply:
Thanks for clarification. I'll change this to `for _, proto := range f.ipProtocols`